app/models/concerns/project_media_cached_fields.rb

```diff
@@ -639,7 +639,9 @@ def recalculate_title
       else
         self.update_column(:custom_title, title)
       end
-      title.to_s
+      # Ensure the field value does not exceed ~32KB (ref: CV2-6395);
+      # truncate at 30_000 to stay safely under 32KB as the maximum size should be 32 * 1024 bytes.
+      title.to_s.truncate(30_000)
```

Comment:
@melsawy I might be misunderstanding, but I tried to run this with the title example you added to the ticket and the original and truncated title were the same size:

```
> title.size
=> 17147
> title2 = title.to_s.truncate(30_000)
=> "https://vm.tiktok.com/ZMSrt46Xj/\\n\\ndata:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAA..."
> title2.length
=> 17147
```

And a couple of questions: Don't we have a maximum character size for titles? If we do, couldn't we truncate using that as a reference?